### PR TITLE
ci: add support for loongarch64-unknown-linux-musl

### PR DESCRIPTION
**Pull request by heiher**
# Description

1. Fix libc-tests for `loongarch64-unknown-linux-musl`.
2. Add support for `loongarch64-unknown-linux-musl` target in CI.

# Sources

None.

# Checklist

- [x] Relevant tests in `libc-test/semver` have been updated
- [x] No placeholder or unstable values like `*LAST` or `*MAX` are
  included (see [#3131](https://github.com/rust-lang/libc/issues/3131))
- [x] Tested locally (`cd libc-test && cargo test --target mytarget`);
  especially relevant for platforms that may not be checked in CI

### CODE REVIEW

File: libc-test/build.rs [hunk 1]
```diff
@@ -4476,14 +4476,14 @@ fn test_linux(target: &str) {
         (struct_ == "iw_encode_ext" && field == "key") ||
         // the `tcpi_snd_rcv_wscale` map two bitfield fields stored in a u8
         (struct_ == "tcp_info" && field == "tcpi_snd_rcv_wscale") ||
-        // the `tcpi_delivery_rate_app_limited` field is a bitfield on musl
-        (musl && struct_ == "tcp_info" && field == "tcpi_delivery_rate_app_limited") ||
-        // the `tcpi_fast_open_client_fail` field is a bitfield on musl
-        (musl && struct_ == "tcp_info" && field == "tcpi_fast_open_client_fail") ||
```
Comment (tgross35) [inline libc-test/build.rs:4482]: Any idea why these aren't a problem anymore?

File: src/unix/linux_like/linux/musl/mod.rs [hunk 2]
```diff
@@ -408,8 +410,10 @@ s! {
         pub tcpi_dsack_dups: u32,
         pub tcpi_reord_seen: u32,
         // FIXME(musl): to uncomment once CI musl is updated
```
Comment (tgross35) [inline src/unix/linux_like/linux/musl/mod.rs:412]:
```suggestion
        // FIXME(musl): enable on all targets once CI musl is updated
```

File: src/unix/linux_like/linux/musl/mod.rs [hunk 3]
```diff
@@ -356,13 +356,15 @@ s! {
         pub tcpi_options: u8,
         /*
          * FIXME(musl): when musl headers are more up to date
+        */
```
Comment (tgross35) [inline src/unix/linux_like/linux/musl/mod.rs:359]:
```suggestion
        // FIXME(musl): enable on all targets once musl headers are more up to date
```

File: libc-test/build.rs (hunk 1 above)
Comment (heiher) [reply libc-test/build.rs:4482]: These fields from newer version of musl are currently only available on the loongarch64 bindings.

File: src/unix/linux_like/linux/musl/mod.rs (hunk 3 above)
Comment (heiher) [reply src/unix/linux_like/linux/musl/mod.rs]: All done. Thanks.

**Review by tgross35 — APPROVED** [review]
LGTM. I'm trying to land some large formatting cleanup in https://github.com/rust-lang/libc/pull/4107, please rebase after that merges.

**Pull request merged**